packages/editor/web/src/RichContentEditor/createPlugins.ts

```diff
@@ -42,7 +42,13 @@ const createPlugins = ({
   plugins?: CreatePluginFunction[];
   context: EditorContextType;
   commonPubsub: Pubsub;
-}) => {
+}): {
+  pluginInstances: unknown[];
```

Comment:
I suggest
```tsx
pluginInstances: (EditorPlugin | ReturnType<CreatePluginFunction>)[]
```

Or more specifically `EditorPlugin & { decorator }` for the first part

Reply:
Not all plugins has & { decorator },
I can go with
```suggestion
  pluginInstances: (
    | EditorPlugin
    | (EditorPlugin & { decorator })
    | ReturnType<CreatePluginFunction>
  )[];
```
Or with your first suggestion.

Reply:
Looks good, I'd make it optional like this

```tsx
pluginInstances: (
    | (EditorPlugin & { decorator?: DraftDecorator })
    | ReturnType<CreatePluginFunction>
  )[]
```

Reply:
Great! thanks.